modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/hooks/useId.ts

```diff
@@ -13,12 +13,19 @@
  */
 
 import {useMemo} from 'react';
+
+// @ts-ignore
+
 import {v4 as uuidv4} from 'uuid';
 
 const PREFIX = uuidv4();
 
 let nextId = 0;
 
-export function useId() {
+/**
+ * Hook to generate unique IDs.
+ */
+
+export default function useId() {
```

Comment:
If the intent is generate a unique identifier, can we simplify this to just
```js
return useMemo(() => uuidv4(), []);
```
This would make `nextId` unnecessary, unless it has some purpose other than uniquness. Is there are reason for `useId_` prefix?

Does it make sense to even have this in a hook or a utility? Why not just inline it?
```js
const uuid = useMemo(() => uuidv4(), []);
```

I guess I see two small reasons why:
- It makes the code a bit more readable.
- We may want to swap implementations at some point, `uuid` is a third party library after all.

If we do want to create it, would it make sense to name the hook a bit more descriptively, something like `useUUID()` or `useUniqueId()`?

Reply:
[Hi @markocikos, thanks for reviewing this PR. Regarding your comments:

> This would make nextId unnecessary, unless it has some purpose other than uniquness. Is there are reason for useId_ prefix?

I think we can remove `nextId`. It's used because `uuidv4` was not used before, but right now it is no longer needed.
We use `useId_` because sometimes ids that don't start with a character cause problems https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id

> Does it make sense to even have this in a hook or a utility? Why not just inline it?
> If we do want to create it, would it make sense to name the hook a bit more descriptively, something like useUUID() or useUniqueId()?

We prefer to leave it as a hook and call it useId to maintain direct compatibility with React, since [useId hook](https://react.dev/reference/react/useId) was included in version 18 and if at any time we think about updating it, the change would be direct.

Reply:
> We use useId_ because sometimes ids that don't start with a character cause problems

Oh that's a good point! We can maybe make it a bit more generic, like `lr-` or `l-` prefix (L for Liferay :slightly_smiling_face: )

> We prefer to leave it as a hook and call it useId to maintain direct compatibility with React, since [useId hook](https://react.dev/reference/react/useId) was included in version 18 and if at any time we think about updating it, the change would be direct.

Also a good point! This will make it much easier to migrate later. :+1:

Reply:
> Oh that's a good point! Maybe we can only make the prefix a bit more generic, like lr- or l- prefix (L for Liferay 🙂 )

Yes, I think it's a good option, what do you think `lfr`? thus we maintain compatibility with other parts of the portal.